.github/workflows/build.yml

```diff
@@ -245,7 +262,140 @@ jobs:
         env:
           PLATFORMS: windows/amd64
           CACHE_DIR: /tmp/.buildkit-cache/${{ env.CACHEKEY_BINARIES }}
+
+  release-base:
+    runs-on: ubuntu-latest
+    outputs:
+      tag: ${{ steps.prep.outputs.tag }}
+      push: ${{ steps.prep.outputs.push }}
+    steps:
+      - name: Prepare
+        id: prep
+        run: |
+          TAG=local
+          PUSH=false
+          if [ "${{ github.event_name }}" = "schedule" ]; then
+            TAG=nightly
+            PUSH=push
+          elif [[ $GITHUB_REF == refs/tags/* ]]; then
+            TAG=${GITHUB_REF#refs/tags/}
+            PUSH=push
+          elif [[ $GITHUB_REF == refs/heads/* ]]; then
+            TAG=$(echo ${GITHUB_REF#refs/heads/} | sed -r 's#/+#-#g')
+            PUSH=push
+          fi
+          PUSH="" # FIXME: Remove when migration to GitHub Actions completed
+          echo ::set-output name=tag::${TAG}
+          echo ::set-output name=push::${PUSH}
+
+  image:
+    runs-on: ubuntu-latest
+    needs: [release-base, test, cross]
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - target-stage: ''
+          - target-stage: rootless
+    steps:
       -
-        name: Dump context
-        if: always()
-        uses: crazy-max/ghaction-dump-context@v1
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Cache ${{ env.CACHEKEY_BINARIES }}
```

Comment:
images themselves do not seem to be cached at all. rootless in qemu based and take 25min to build.

Reply:
Might even make sense to just cache the base of the images stages without the new binaries.